packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -639,31 +678,63 @@ class _TextSelectionHandleOverlayState
       point.dy.clamp(0.0, viewport.height),
     );
 
+    final Offset handleAnchor = widget.selectionControls.getHandleAnchor(
+      type,
+      widget.renderObject.preferredLineHeight,
+    );
+    final Size handleSize = widget.selectionControls.getHandleSize(
+      widget.renderObject.preferredLineHeight,
+    );
+    final Rect handleRect = Rect.fromLTWH(
+      // Put handleAnchor on top of point
+      point.dx - handleAnchor.dx,
+      point.dy - handleAnchor.dy,
+      handleSize.width,
+      handleSize.height,
+    );
+
+    // Make sure the GestureDetector is big enough to be easily interactive.
+    final Rect interactiveRect = getInteractiveRect(handleRect);
```

Comment:
I think you can dispense with all of this:

```
RelativeRect padding = RelatveRect.fromRect(rect, Rect.fromCircle(center: rect.center, radius: kMinInteractiveSize));
```
And then:
```
padding: EdgeInsets.only(
  left: math.max(padding.left, 0),
  top: math.max(padding.top, 0),
  right: math.max(padding.right, 0),
  bottom: math.max(padding.bottom, 0),
),
```

Reply:
I tried this out and ended up going with something pretty similar.